packages/builders/src/components/button/SKUIdButton.ts

```diff
@@ -0,0 +1,26 @@
+import type { APIButtonComponentWithSKUId, Snowflake } from 'discord-api-types/v10';
+import { ButtonStyle, ComponentType } from 'discord-api-types/v10';
+import { ButtonBuilder } from './Button.js';
+
+/**
+ * A builder that creates API-compatible JSON data for SKU ID buttons.
+ */
+export class SKUIdButtonBuilder extends ButtonBuilder<APIButtonComponentWithSKUId> {
```

Comment:
I think this should be called `PremiumButtonBuilder` to match the style name. (TSDoc, filename, and methods for adding this builder need to be updated)